block/manager.go

```diff
@@ -595,7 +595,9 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 	}
 
 	// SaveValidators commits the DB tx
+	m.lastStateMtx.Lock()
```

Comment:
I think we should also clean up the `Manager` struct to make it more explicit all the functions that should be performed under this mutex.

The mutex also isn't even next to the field that it is protecting. 

We should at minimum group these fields with a comment explaining the connection:
```go
lastState types.State
store store.Store
lastStateMtx *sync.Mutex
```

Are there other fields that cold also be consider?

Doing a quick code search I found two other places that this mutex is used and I think we should clean up those spots too.

First, `IsProposer` is an exported method that modifies a field protected by a mutex but it doesn't hold the mutex. It is always best to put the mutex inside the method that manipulates the controlled field if possible. 

Second is the place that is involved with this Data Race
```go
		m.lastStateMtx.Lock()
		m.lastState = newState
		m.lastStateMtx.Unlock()
		err = m.store.UpdateState(m.lastState)
```
We should hold the lock while we update the state in `UpdateState` as well, otherwise we are potentially updating it with a different state than expected. 

Again i would recommend a helper
```go
func (m *Manager) updateState(newState types.State) error {
  m.lastStateMtx.Lock()
  defer m.lastStateMtx.Unlock()
  err = m.store.UpdateState(newState)
  if err {}
  m.lastState = newState
  return nil
}
```